crypto/s2n_hash.c

```diff
@@ -299,8 +342,10 @@ static int s2n_evp_hash_init(struct s2n_hash_state *state, s2n_hash_algorithm al
         return S2N_SUCCESS;
     }
 
-    POSIX_ENSURE_REF(s2n_hash_alg_to_evp_md(alg));
-    POSIX_GUARD_OSSL(EVP_DigestInit_ex(state->digest.high_level.evp.ctx, s2n_hash_alg_to_evp_md(alg), NULL), S2N_ERR_HASH_INIT_FAILED);
+    const EVP_MD *md = s2n_hash_alg_to_evp_md(alg);
+    POSIX_ENSURE(md, S2N_ERR_HASH_INVALID_ALGORITHM);
+    POSIX_GUARD_OSSL(EVP_DigestInit_ex(state->digest.high_level.evp.ctx, md, NULL),
+            S2N_ERR_HASH_INIT_FAILED);
```

Comment:
This doesn't cause more/less failures, it just makes the reason for failures more explicit. It seemed worth adding the S2N_ERR_HASH_INVALID_ALGORITHM here since hash_init is the place we actually kind of expect to discover a hash isn't allowed.